libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteRefreshOptions.kt

```diff
@@ -99,6 +99,6 @@ class RouteRefreshOptions private constructor(
     }
 
     private companion object {
-        private val MINIMUM_REFRESH_INTERVAL = TimeUnit.SECONDS.toMillis(30)
+        private val MINIMUM_REFRESH_INTERVAL = TimeUnit.SECONDS.toMillis(10)
```

Comment:
Why does the interval value need to change?

Reply:
It doesn't need to, really. Thanks for pinging.

- Idling resources timeout so we just need to [update the IdlingPolicy](https://developer.android.com/reference/androidx/test/espresso/IdlingPolicies) for the test
- Instrumentation tests are taking 10 minutes to pass. so 30 seconds is well below

We were concerned about running time in this [previous pull request](https://github.com/mapbox/mapbox-navigation-android/pull/4375#discussion_r631738092) and i didn't give it much effort. So this test I gave it some more thought.

Reply:
Made a nice `IdlingPolicyTimeoutRule`. So when we run into these timeout issues it's a 2 liner.

``` kotlin
@get:Rule
    val idlingPolicyRule = IdlingPolicyTimeoutRule(35, TimeUnit.SECONDS)
```

Reply:
The problem with this approach is that it doesn't scale.

If we wanted to have only 10 tests like these that verify route refresh or alternative route lines we'd already be looking at 5 minutes longer execution time for our CI pipeline which not only means costs but also frustration. And this is already assuming a very generous minimal interval of 30 seconds, which we've previously set to 2 or 5 minutes for features like these.

@kmadsen I see you've explored device time-shifting, but did you try that with `adb`? Is that possible?

Reply:
> The problem with this approach is that it doesn't scale.

Instrumentation tests can run in parallel though, right? Is 30 seconds beyond our limit? The tests are currently taking 10 minutes so the 30 seconds seems small. Here's a run from the main line [1 min 23 sec](https://console.firebase.google.com/u/0/project/mapbox-navigation-android/testlab/histories/bh.54eb2dd57e959221/matrices/4619913019565188462/executions/bs.8a8b3d2f5942c626) running time. Here's a run from this branch [1 min 36 sec](https://console.firebase.google.com/u/0/project/mapbox-navigation-android/testlab/histories/bh.54eb2dd57e959221/matrices/8186676078878841656/executions/bs.1fe68d4177711a33). It seems we should be considering the scaling solutions with Firebase too.

> I see you've explored device time-shifting, but did you try that with adb? Is that possible?

An `adb` time shifting approach is worth looking into. The conclusion was that we can continue optimizations for tests. But am not considering this change significant.

In my opinion, we will want to be able to control the idling policy. And this isn't a big offender, so the change is good. What do you think @LukasPaczos ?

Reply:
Feel free to jump in and try the adb time shifting! I spent a day messing around on these things 😄

Reply:
Another reason I prefer to get these tests in is because the route refresh [tests are actively preventing us from breaking the feature](https://github.com/mapbox/mapbox-navigation-android/issues/4429#issuecomment-846249167).

Thinking about `adb` time shifting. I can see you can change the display time, but wonder if you can change the monotonic [elapsedRealtimeNanos](https://developer.android.com/reference/android/os/SystemClock#elapsedRealtimeNanos()) 🤔

Reply:
> Is 30 seconds beyond our limit?

It definitely is. I know we are currently behind in instrumentation tests, but 30s per test would not be acceptable, I would target 3-5s as a maximum. We can always introduce a nightly build, but that shouldn't be targeted at testing a singular feature like "if route refresh is parsed" correctly which conceptually is a "small" test and should be run for each contribution. Nightly tests should involve bigger and longer scenarios.

> but wonder if you can change the monotonic

I was afraid that the time from boot wouldn't be accessible, and after all, it makes sense that it isn't. Only the wall clock might accessible in some nasty ways but that won't help us.

> Instrumentation tests can run in parallel though, right?

Test Orchestrator can do that, but you'd need access to multiple devices in the first place. Looks like Firebase has a feature called ["test sharding"](https://firebase.google.com/docs/test-lab/android/instrumentation-test#sharding) which does just that. We'd have to test it out.